### ISSUE TITLE
Make treatment of delimiters in missing/strjoin compatible with Matlab

### ISSUE THREAD

**Issue by JohannesPfeifer** [DynareTeam/dynare#1106]
The Dynare version returns for ```strjoin({'aa','bb'},'\n')```
```ans =aa\nbb```
while Matlab returns
```
ans =
aa
bb
``` 
The same holds for other delimiters like the tabulator ```\t``` that are treated like literal characters.

**Comment by MichelJuillard** [comment]
Be careful, this may have implications for the manipulation of TeX strings. In which cases is missing/strjoin called?

**Comment by JohannesPfeifer** [comment]
@MichelJuillard I thought the files in the ```matlab/missing``` folders are only added to the path if the Matlab version is not detected. I have never experienced problems with the proprietary Matlab version of ```strjoin``` so I presume that having full compatibility of our version with it should not create any problems.

**Comment by MichelJuillard** [comment]
Let see what @houtanb and @stepan-a say

**Comment by houtanb** [comment]
In fact, I only use it in reporting to display error messages. I took the open source version of the file that I found at the time and modified it to work with Matlab. I only intersperse commas in a cell arrays, so the escape characters don't factor in.

So, in short, this is a bug we won't ever encounter.

If you'd like to fix this to work with escape characters because you are planning to implement code using it, feel free to use the version of `strjoin.m` provided in Octave 4.0: https://gist.github.com/houtanb/1677b8cefc4b29c16c9b You'll have to modify some of the syntax to work with Matlab. Otherwise, close this ticket.
